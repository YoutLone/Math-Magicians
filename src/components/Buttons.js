import React from 'react';
import PropTypes from 'prop-types';

// Button component
function Button({ text, onClick, className }) {
  return (
    <button type="button" className={className} onClick={onClick}>
      {text}
    </button>
  );
}

Button.propTypes = {
  text: PropTypes.string.isRequired,
  onClick: PropTypes.func,
  className: PropTypes.string,
};

Button.defaultProps = {
  onClick: () => {},
  className: '',
};

export default Button;
