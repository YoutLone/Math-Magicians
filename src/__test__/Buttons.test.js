import React from 'react';
import renderer from 'react-test-renderer';
import Buttons from '../components/Buttons';

describe('Buttons', () => {
  test('should render the Button component', () => {
    const handleInput = jest.fn();

    const component = renderer.create(
      <Buttons text="1" className="my-class" clickEvent={handleInput} />,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    tree = component.root;
    expect(tree.findByType('button').props.children).toBe('1'); // Updated assertion
  });
});
