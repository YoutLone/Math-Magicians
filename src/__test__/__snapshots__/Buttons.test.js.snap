// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Buttons should render the Button component 1`] = `
<button
  className="my-class"
  onClick={[Function]}
  type="button"
>
  1
</button>
`;
